src/inc/eventtracebase.h

```diff
@@ -100,7 +100,7 @@ enum EtwThreadFlags
 #define ETWFireEvent(EventName)
 
 #define ETW_TRACING_INITIALIZED(RegHandle) (TRUE)
-#define ETW_EVENT_ENABLED(Context, EventDescriptor) (EventPipeHelper::Enabled() || EventEnabled##EventDescriptor())
+#define ETW_EVENT_ENABLED(Context, EventDescriptor) (EventPipeHelper::Enabled() || XplatEventLogger::IsEventLoggingEnabled())
```

Comment:
I went to the bathroom and realised I actually do have a question here. Why change this here AND add this check in `genEventing.py`? Wouldn't just the second be enough, and wouldn't removing the `EventEnabled##EventDescriptor` here result in a more general check on x64?

Reply:
ETW_EVENT_ENABLED is not always used - it is only used in a subset of cases.  The change in genEventing.py handles the case where EventEnabled* is called.

Removing EventEnabled##EventDescriptor does make the check more generic on some platforms depending on which version of LTTng we build against.  This changes makes sure that you are forced to go through the environment variable check.  Going forward, we should switch to building against LTTng 2.7+ for all architectures and then we won't need the environment variable.  Right now, arm builds on ubuntu 14.04 while x64 builds on RHEL (not sure which version) and the version of LTTng that we build against is different for each.